dhall/src/Dhall/Diff.hs

```diff
@@ -745,7 +754,22 @@ diffAnnotatedExpression l r =
     diffOperatorExpression l r
 
 diffOperatorExpression :: (Eq a, Pretty a) => Expr Void a -> Expr Void a -> Diff
-diffOperatorExpression = diffOrExpression
+diffOperatorExpression = diffImportAltExpression
+
+diffImportAltExpression :: (Pretty a, Eq a) => Expr Void a -> Expr Void a -> Diff
+diffImportAltExpression l@(ImportAlt {}) r@(ImportAlt {}) =
+    enclosed' "    " (operator "?" <> "  ") (docs l r)
+  where
+    docs (ImportAlt aL bL) (ImportAlt aR bR) =
+        Data.List.NonEmpty.cons (diffOrExpression aL aR) (docs bL bR)
```

Comment:
Is the use of `diffOrExpression` correct here? In general these functions seem to use the diff for next-higher precedence here, but a lot of them also deviate.

Reply:
Your use of `diffOrExpression` here is correct and the deviations from this pattern are bugs